manifests/init.pp

```diff
@@ -131,11 +132,88 @@
   Boolean                       $use_internet_repo       = false,
   String                        $version                 = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
   String                        $rpm_iteration           = '1',
-  Boolean                       $simp_dashboards         = false
-) inherits ::simp_grafana::params {
+  Boolean                       $simp_dashboards         = false,
+  String                        $admin_pw                = passgen('grafana'),
+  String                        $app_pki_dir             = '/etc/pki/simp_apps/grafana/x509',
+  String                        $app_pki_key             = "${app_pki_dir}/private/${facts['fqdn']}.pem",
+  String                        $app_pki_cert            = "${app_pki_dir}/public/${facts['fqdn']}.pub",
+ ) inherits ::simp_grafana::params {
 
-  $merged_cfg = deep_merge($::simp_grafana::params::cfg, $cfg)
-  $merged_ldap_cfg = deep_merge($::simp_grafana::params::ldap_cfg, $ldap_cfg)
+  # Static defaults
+  $_cfg = {
+    server       => {
+      http_port => 8443,
+      protocol  => 'https',
+      cert_file => $app_pki_cert,
+      cert_key  => $app_pki_key,
+    },
+    security     => {
+      admin_password   => $admin_pw,
+      disable_gravatar => true,
+    },
+    users        => {
+      allow_sign_up    => false,
+      allow_org_create => true,
+      auto_assign_org  => true,
+    },
+    'auth.basic' => { enabled => false },
+    'auth.ldap'  => { enabled => $ldap },
+    #Allows SIMP dashboards to be read from the file system
+    'dashboards.json' => { enabled => true },
+    analytics   => { reporting_enabled => false },
+    snapshot    => { external_enabled => false },
+  }
+
+  if ($ldap) {
+
+    $ldap_group_mapping_defaults = [
```

Comment:
You may want to break this entire block of logic out into a Puppet function. I see why it's here but I think that a function called simp_grafana::ldap_server_defaults() that does all of the work might make things easier for users since they can look at the function very specifically if that's what they want.

You may also want to make the defaults a parameter in cased someone needs to change the default settings. As it sits, for both `cfg` and `ldap_cfg` the user doesn't have any way to tell us not to use the defaults at all which looks like an oversight in the original design.